### PR TITLE
change type of IdArray from proto to arrow

### PR DESCRIPTION
**Pull request by jiaoew1991**
issue: https://github.com/milvus-io/milvus/issues/16733
Signed-off-by: Enwei Jiao <jiaoew2011@gmail.com>

### CODE REVIEW

**Comment by xiaofan-luan** [comment]
@jiaoew1991 mean while we should add more test cases to improve ut coverage

**Comment by xiaofan-luan** [comment]
> @jiaoew1991 mean while we should add more test cases to improve ut coverage

weird, from the code coverage report ut is fine, probably rebase would help

**Comment by jiaoew1991** [comment]
> > @jiaoew1991 mean while we should add more test cases to improve ut coverage
> 
> weird, from the code coverage report ut is fine, probably rebase would help

It looks like I changed some code that not be covered long time ago 😄，I will add some testcases

**Comment by jiaoew1991** [comment]
> > > @jiaoew1991 mean while we should add more test cases to improve ut coverage
> > 
> > 
> > weird, from the code coverage report ut is fine, probably rebase would help
> 
> It looks like I changed some code that not be covered long time ago 😄，I will add some testcases

@xiaofan-luan the coverage is passed 😄

**Comment by xiaofan-luan** [comment]
> > > > @jiaoew1991 mean while we should add more test cases to improve ut coverage
> > > 
> > > 
> > > weird, from the code coverage report ut is fine, probably rebase would help
> > 
> > 
> > It looks like I changed some code that not be covered long time ago 😄，I will add some testcases
> 
> @xiaofan-luan the coverage is passed 😄

Thanks my man, I'm working on reviewing it and the overall looks good~

**Comment by xiaofan-luan** [comment]
We will wait until 2.1 branch created and we merge this into master

File: internal/core/src/query/visitors/ExecExprVisitor.cpp [hunk 1]
```diff
@@ -764,20 +764,22 @@ ExecExprVisitor::ExecTermVisitorImpl(TermExpr& expr_raw) -> BitsetType {
...
                 for (const auto& id : expr.terms_) {
-                    dst_ids->add_data((int64_t&)id);
+                    builder.Append(id);
                 }
+                id_array = builder.Finish().ValueOrDie();
                 break;
             }
             case DataType::VARCHAR: {
-                auto dst_ids = id_array->mutable_str_id();
+                auto builder = arrow::StringBuilder();
```
Comment (jiaoew1991) [inline internal/core/src/query/visitors/ExecExprVisitor.cpp:778]: This part of the code is redundant

there is a method  `ExecExprVisitor::ExecTermVisitorImpl<std::string>` below, so this part will never be used 😅

**Pull request closed without merging**